t/167-server-rewite.t

```diff
@@ -0,0 +1,223 @@
+
+#log_level("info");
+#no_long_string();
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: server_rewrite_by_lua_block in http
+--- http_config
+	server_rewrite_by_lua_block {
+		ngx.ctx.a = "server_rewrite_by_lua_block in http"
+	}
+--- config
+   	location /lua {
+       	content_by_lua_block {
+           	ngx.say(ngx.ctx.a)
+		   	ngx.log(ngx.INFO, ngx.ctx.a)
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+server_rewrite_by_lua_block in http
+--- error_log
+server_rewrite_by_lua_block in http
+--- no_error_log
+[error]
+
+=== TEST 2: server_rewrite_by_lua_block in server
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "server_rewrite_by_lua_block in server")
+	}
+   	location /lua {
+       	content_by_lua_block {
+       	    ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+server_rewrite_by_lua_block in server
+--- no_error_log
+[error]
+
+
+=== TEST 3: server_rewrite_by_lua_block overwrite by server
+--- http_config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "server_rewrite_by_lua_block in http")
+	}
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "server_rewrite_by_lua_block in server")
+	}
+   	location /lua {
+       	content_by_lua_block {
+       	    ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+server_rewrite_by_lua_block in server
+--- no_error_log
+[error]
+
+
+=== TEST 4: sleep
+--- config
+	server_rewrite_by_lua_block {
+		ngx.sleep(0.001)
+		ngx.log(ngx.INFO, "server_rewrite_by_lua_block in server")
+	}
+   	location /lua {
+       	content_by_lua_block {
+       	    ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+server_rewrite_by_lua_block in server
+--- no_error_log
+[error]
+
+
+=== TEST 5: ngx.exit(ngx.OK)
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "ngx.exit")
+        ngx.exit(ngx.OK)
+	}
+   	location /lua {
+       	content_by_lua_block {
+           ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+ngx.exit
+--- no_error_log
+[error]
+
+
+=== TEST 6: ngx.exit(503)
+--- config
+	server_rewrite_by_lua_block {
+        ngx.exit(503)
+	}
+   	location /lua {
+       	content_by_lua_block {
+			ngx.log(ngx.ERR, "content_by_lua")
+        	ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- error_code: 503
+--- no_error_log
+[error]
+
+
+=== TEST 7: subrequests
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "is_subrequest:", ngx.is_subrequest)
+	}
+
+   	location /lua {
+       	content_by_lua_block {
+			local res = ngx.location.capture("/sub")
+			ngx.print(res.body)
+		}
+   	}
+
+	location /sub {
+       	content_by_lua_block {
+           	ngx.say("OK")
+       	}
+   	}
+
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+is_subrequest:true
+is_subrequest:false
+--- no_error_log
+[error]
+
+
```

Comment:
You should use our reindex script from the openresty-devel-utils repo to reformat your test files.

Reply:
done